### ISSUE TITLE
No NFTs for Sale

### ISSUE THREAD

**Issue by teambrixmusic** [bitclout/frontend#348]
I have an NFT for sale. All along it's said I don't have any for sale. I thought perhaps the blockchain would catch up or it was a small issue but I don't see the problem with other accounts. So I thought I'd report it.

![IDK](https://user-images.githubusercontent.com/82254764/127730849-e2d8f431-59f9-4a42-8412-b92d3765ae60.jpg)

**Comment by tijno** [comment]
Whats the link to the nft ?

**Comment by tijno** [comment]
Did some digging to help move this along hopefully.

Here is one of your NFTs as an example:

https://bitclout.com/nft/8265b2cf269a5e3068d3c7fe1f0c06d56c5e7fc63280784a6f9f56475e7e5e9d?tab=posts

When making a call to `get-nfts-for-user` with `IsForSale: False` the response includes the above NFT:

```json
"5a1e3cf992c66df0dfddf374b7f3f93e2ed6c825911bbeed9fec8c97f1f04e7f": {
      "PostEntryResponse": {
        "PostHashHex": "5a1e3cf992c66df0dfddf374b7f3f93e2ed6c825911bbeed9fec8c97f1f04e7f",
        "PosterPublicKeyBase58Check": "BC1YLjEKniMLzmYA7QURRohZYF4QFZjCWoEqj3Zkf1UsYxULKyWLKYY",
        "ParentStakeID": "",
        "Body": "𝘛𝘩𝘦 𝘜𝘯𝘬𝘯𝘰𝘸𝘯 𝘊𝘰𝘭𝘭𝘦𝘤𝘵𝘪𝘰𝘯\n\n〰️【 ﻿𝐑𝐢𝐩𝐩𝐥𝐞𝐬 𝐨𝐟 𝐓𝐢𝐦𝐞 】⌛\n\n2048 x 2048 | 50 Editions\n\n#unknownnft #cloutnft #abstract #time #gan #ai",
        "ImageURLs": [
          "https://images.bitclout.com/8654fe222e994726900fab67f7d8983af8d487cf3f53843421b770da53e89c7f.webp"
        ],
        "RecloutedPostEntryResponse": null,
        "CreatorBasisPoints": 1000,
        "StakeMultipleBasisPoints": 12500,
        "TimestampNanos": 1627527716543606800,
        "IsHidden": false,
        "ConfirmationBlockHeight": 46734,
        "InMempool": false,
        "StakeEntry": {
          "TotalPostStake": 0,
          "StakeList": []
        },
        "StakeEntryStats": {
          "TotalStakeNanos": 0,
          "TotalStakeOwedNanos": 0,
          "TotalCreatorEarningsNanos": 0,
          "TotalFeesBurnedNanos": 0,
          "TotalPostStakeNanos": 0
        },
        "ProfileEntryResponse": {
          "PublicKeyBase58Check": "BC1YLjEKniMLzmYA7QURRohZYF4QFZjCWoEqj3Zkf1UsYxULKyWLKYY",
          "Username": "UnknownNFT",
          "Description": "GAN Artist // NFT + Coin Collector // Crypto\n\nHigh-quality, one-of-a-kind NFTs 🎨\n\n🧊 @coldstorage 🔒\n🏴‍☠️ @cloutmafia 🕶\n📚 @cloutquote 💬\n🔑  @room21 🚪\n\nhttps://y.at/👽🚀🌕👽",
          "IsHidden": false,
          "IsReserved": false,
          "IsVerified": false,
          "Comments": null,
          "Posts": null,
          "CoinEntry": {
            "CreatorBasisPoints": 1100,
            "BitCloutLockedNanos": 20153596487,
            "NumberOfHolders": 899,
            "CoinsInCirculationNanos": 27152458927,
            "CoinWatermarkNanos": 27723542168
          },
          "CoinPriceBitCloutNanos": 2226715291,
          "StakeMultipleBasisPoints": 12500,
          "StakeEntryStats": {
            "TotalStakeNanos": 0,
            "TotalStakeOwedNanos": 0,
            "TotalCreatorEarningsNanos": 0,
            "TotalFeesBurnedNanos": 0,
            "TotalPostStakeNanos": 0
          },
          "UsersThatHODL": null
        },
        "Comments": null,
        "LikeCount": 21,
        "DiamondCount": 11,
        "PostEntryReaderState": null,
        "InGlobalFeed": false,
        "IsPinned": false,
        "PostExtraData": {},
        "CommentCount": 13,
        "RecloutCount": 5,
        "QuoteRecloutCount": 12,
        "ParentPosts": null,
        "IsNFT": true,
        "NumNFTCopies": 50,
        "NumNFTCopiesForSale": 26,
        "HasUnlockable": true,
        "NFTRoyaltyToCreatorBasisPoints": 1500,
        "NFTRoyaltyToCoinBasisPoints": 1000,
        "DiamondsFromSender": 0
      },
      "NFTEntryResponses": [
        {
          "OwnerPublicKeyBase58Check": "BC1YLhTk3qzocZRDAAHyFvy7PME3xvPZyJn3eY2tfjGa8GmNwheU8Xv",
          "SerialNumber": 33,
          "IsForSale": false,
          "MinBidAmountNanos": 0,
          "LastAcceptedBidAmountNanos": 10559662,
          "HighestBidAmountNanos": 0,
          "LowestBidAmountNanos": 0
        }
      ]
    }
    ```
    
    `IsForSale:true` requests return empty NFTsMap.
    
    So the issue is in the backend not returning the available ForSale serials I think.
    
    Dont know enough yet about how multi serial NFTs are handled - but I believe when minting them it loops through them one by one. 
    
    Will check more.

**Comment by teambrixmusic** [comment]
> Whats the link to the nft ?

This one: https://bitclout.com/posts/e62ac72dd5b5e55ba62312a8de86857ff6210f73fa6da08dd5fc48b7d17e0c76?tab=posts

What's sad I messaged 100+ coinholders and told them I minted my first NFT. And I didn't link it as I didn't think I had to (links aren't linkable in messages which is another subject.)

And I have a slew of folks who went looking for it and couldn't find it. Or said they must have missed it. And I look nuts, annoying and thirsty having to reclout it over and over and link to it. I think that turns some people off too. Looks desperate. It's so embarrassing. Just started messaging people back as I didn't know! And now the top 10 are already pretty much bought. I would have liked my own coinholders to have been able to get a heads up and then easily find it. People move on pretty quickly if things are too difficult.
